tools/autograd/templates/Functions.cpp

```diff
@@ -665,6 +665,18 @@ Tensor kl_div_double_backward_grad_output(const Tensor & grad, const Tensor & in
   return result;
 }
 
+Tensor kl_div_target_backward(Tensor grad_output, Tensor self, Tensor target, bool size_average, bool reduce) {
+  if (!reduce) {
+    return grad_output.mul(target.log() + at::ones_like(target) - self).masked_fill_(target == 0, 0.);
+  }
+
+  auto norm = at::ones_like(target);
+  if (size_average) {
+    norm = norm.div(norm.numel());
+  }
+  return grad_output.mul(target.log() + at::ones_like(target) - self).mul(norm).masked_fill_(target == 0, 0.);
```

Comment:
Is it possible to avoid the allocation of a `norm` tensor? You might be able to allocate a scalar tensor for it, or just use a scalar entirely: (does this work?)
```
(grad_output.mul(target.log() + at::ones_like(target) - self) / grad_output.numel()).masked_fill(...)
```